mailbox/event/event-cassandra/src/test/java/org/apache/james/mailbox/events/CassandraEventDeadLettersGroupDAOTest.java

```diff
@@ -0,0 +1,74 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.apache.james.mailbox.events.EventDeadLettersContract.GROUP_A;
+import static org.apache.james.mailbox.events.EventDeadLettersContract.GROUP_B;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.backends.cassandra.CassandraClusterExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class CassandraEventDeadLettersGroupDAOTest {
+
+    @RegisterExtension
+    static CassandraClusterExtension cassandraClusterExtension = new CassandraClusterExtension(CassandraEventDeadLettersModule.MODULE);
+
+    private CassandraEventDeadLettersGroupDAO groupDAO;
+
+    @BeforeEach
+    void setUp(CassandraCluster cassandraCluster) {
+        groupDAO = new CassandraEventDeadLettersGroupDAO(cassandraCluster.getConf());
+    }
+
+    @Test
+    void retrieveAllGroupsShouldReturnEmptyWhenDefault() {
+        assertThat(groupDAO.retrieveAllGroups()
+                .collectList().block())
+            .isEmpty();
+    }
+
+    @Test
+    void retrieveAllGroupsShouldReturnStoredGroups() {
+        groupDAO.storeGroup(GROUP_A).block();
+        groupDAO.storeGroup(GROUP_B).block();
+        groupDAO.storeGroup(GROUP_B).block();
```

Comment:
It's not clear in this test that storing a group is idempotent.

You can add another test dedicated to that point.

Reply:
srr, I am newbie of Cassandra, I want to test the `removeGroup` method with duplicated case. But now, we don't have the `removeGroup`. I deleted.